powershell/cmdlets/class.ts

```diff
@@ -1260,6 +1268,22 @@ export class CmdletClass extends Class {
             yield '    return;';
             yield '}';
           });
+          yield Else(function * () {
+            yield 'if (data.ResponseMessage is System.Net.Http.HttpResponseMessage response)';
+            yield '{';
+            yield '    int delay = (int)(response.Headers.RetryAfter?.Delta?.TotalSeconds ?? 30);';
+            yield '    WriteDebug($"Delaying {delay} seconds before polling.");';
+            yield '    for (var now = 0; now < delay; ++now)';
+            yield '    {';
+            // hardcode id = 1 because there is no need for nested progress bar
+            yield `        WriteProgress(new global::System.Management.Automation.ProgressRecord(1, "In progress", "Checking operation status")`;
+            yield '        {';
+            yield '            PercentComplete = now * 100 / delay';
+            yield '        });';
+            yield `        await global::System.Threading.Tasks.Task.Delay(1000, ${token.use});`;
```

Comment:
1. Except for setting the PercentComplete, does powershell support to set a duration and then automatically go when time passes
2. Ideally we should add IsCancellation check after the await

Reply:
1. The [WriteProgress API](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.progressrecord?view=powershellsdk-7.0.0) can't work that way.
2. Yes but this case is different. I pass the token **into** the `Delay()` API so that user can cancel at any time.